usage/test-framework/src/test/java/org/apache/brooklyn/test/framework/TestSensorTest.java

```diff
@@ -0,0 +1,270 @@
+        boolean sensorTestFail = false;
+        final String sensorValue = String.format("%s%s%s", UUID.randomUUID().toString(), System.currentTimeMillis(), UUID.randomUUID().toString());
+
+        //Add Sensor Test for STRING sensor
+        app.createAndManageChild(EntitySpec.create(TestSensor.class)
+                .configure(TestSensor.TARGET_ENTITY, app)
+                .configure(TestSensor.SENSOR_NAME, STRING_SENSOR.getName())
+                .configure(TestSensor.ASSERTIONS, ImmutableMap.of("regex", String.format(".*%s.*", UUID.randomUUID().toString()))));
+
+        //Set STRING sensor
+        app.sensors().set(STRING_SENSOR, sensorValue);
+
+
+        try {
+            app.start(ImmutableList.of(loc));
+        } catch (final PropagatedRuntimeException pre) {
+            final AssertionError assertionError = Exceptions.getFirstThrowableOfType(pre, AssertionError.class);
+            assertThat(assertionError).isNotNull().as("An assertion error should have been thrown");
+            sensorTestFail = true;
+        } finally {
+            assertThat(sensorTestFail).isTrue().as("regex assert should have failed");
+        }
+    }
+
+    @Test
+    public void testAssertRegexOnNullSensor() {
+        boolean sensorTestFail = false;
+        final String sensorValue = String.format("%s%s%s", UUID.randomUUID().toString(), System.currentTimeMillis(), UUID.randomUUID().toString());
+
+        //Add Sensor Test for STRING sensor
+        app.createAndManageChild(EntitySpec.create(TestSensor.class)
+                .configure(TestSensor.TARGET_ENTITY, app)
+                .configure(TestSensor.SENSOR_NAME, STRING_SENSOR.getName())
+                .configure(TestSensor.ASSERTIONS, ImmutableMap.of("regex", String.format(".*%s.*", UUID.randomUUID().toString()))));
+
+        try {
+            app.start(ImmutableList.of(loc));
+        } catch (final PropagatedRuntimeException pre) {
+            final AssertionError assertionError = Exceptions.getFirstThrowableOfType(pre, AssertionError.class);
+            assertThat(assertionError).isNotNull().as("An assertion error should have been thrown");
+            sensorTestFail = true;
+        } finally {
+            assertThat(sensorTestFail).isTrue().as("regex assert should have failed");
+        }
+    }
+
+
+    @Test
+    public void testAssertRegexOnNonStringSensor() {
+        //Add Sensor Test for OBJECT sensor
+        app.createAndManageChild(EntitySpec.create(TestSensor.class)
+                .configure(TestSensor.TARGET_ENTITY, app)
+                .configure(TestSensor.SENSOR_NAME, OBJECT_SENSOR.getName())
+                .configure(TestSensor.ASSERTIONS, ImmutableMap.of("regex", ".*TestObject.*id=.*")));
+
+        app.sensors().set(OBJECT_SENSOR, new TestObject());
+
+        app.start(ImmutableList.of(loc));
+
+    }
+
+
+    class TestObject {
+        private final String id;
+
+        public TestObject() {
+            id = UUID.randomUUID().toString();
```

Comment:
FWIW we tend to use `Identifiers.makeRandomId(/* length */ 8)` instead of `UUID` -- no big reason but it makes for more easily distinguishable identifiers, shorter and more chars